smartsim/_core/schemas/dragonRequests.py

```diff
@@ -39,6 +40,55 @@
 class DragonRequest(BaseModel): ...
 
 
+class DragonRunPolicy(BaseModel):
+    """Policy specifying hardware constraints when running a Dragon job"""
+
+    device: t.Literal["cpu", "gpu", ""] = Field(default="")
+    """Specify a device category on which to run the job. Uses system default
+    if not specified"""
+    cpu_affinity: t.List[NonNegativeInt] = Field(default_factory=list)
+    """List of CPU indices to which the job should be pinned"""
+    gpu_affinity: t.List[NonNegativeInt] = Field(default_factory=list)
+    """List of GPU indices to which the job should be pinned"""
+
+    @staticmethod
+    def from_run_args(
+        run_args: t.Dict[str, t.Union[int, str, float, None]]
+    ) -> "DragonRunPolicy":
+        """Create a DragonRunPolicy from a dictionary of run arguments"""
+        features: str = str(run_args.get("node-feature", ""))
+
+        device = ""
+        if "gpu" in features:
+            device = "gpu"
+        elif "cpu" in features:
+            device = "cpu"
+
+        gpu_args = ""
+        if gpu_arg_value := run_args.get("gpu-affinity", None):
+            gpu_args = str(gpu_arg_value)
+
+        cpu_args = ""
+        if cpu_arg_value := run_args.get("cpu-affinity", None):
+            cpu_args = str(cpu_arg_value)
+
+        # run args converted to a string must be split back into a list[int]
+        gpu_affinity = [int(x.strip()) for x in gpu_args.split(",") if x]
+        cpu_affinity = [int(x.strip()) for x in cpu_args.split(",") if x]
+
+        if device == "" and not (cpu_affinity or gpu_affinity):
```

Comment:
Do we need this ``if`` split in the code?  If the "not-included" values are aligned with the ``default_factory``, I'm not sure why we wouldn't just pass them in.  Could we just simplify with the last if statement?  As the method grows, it might be difficult to keep adding if statements.

```python
try:
     return DragonRunPolicy(
                device=device,
                cpu_affinity=cpu_affinity,
                gpu_affinity=gpu_affinity,
            )
        except ValidationError as ex:
            raise SmartSimError("Unable to build DragonRunPolicy") from ex
```

Reply:
Good eye. Not anymore! I [sanity tested with a little test harness](https://gist.github.hpe.com/chris-mcbride/f5567ce36d27e3e1f6f0635fb45193ad) to confirm that empty affinity lists do not negatively impact the policy.